src/drtsans/load.py

```diff
@@ -102,10 +103,27 @@ def _insert_periodic_timeslice_log(
     except AttributeError:
         run_start = sample_logs.start_time.value
 
+    try:
```

Comment:
can you create class property `SampleLogs.full_duration` to return the duration (in seconds) of the whole run, meaning from the beginning to the end of the run? Then you write:
```python
    log = periodic_index_log(
        period=time_period,
        interval=time_interval,
        duration=sample_logs.full_duration,
        run_start=run_start,
        offset=time_offset,
        step=1.0,
```
and we'll be able to reuse somewhere else if necessary